frontera/contrib/messagebus/zeromq/socket_config.py

```diff
@@ -1,29 +1,33 @@
 # -*- coding: utf-8 -*-
+from socket import getaddrinfo, gaierror
 
 
 class SocketConfig(object):
-
-    hostname = None
-    base_port = None
-
     def __init__(self, hostname, base_port):
-        self.hostname = hostname
-        self.base_port = base_port
+        try:
+            addr_tuple = getaddrinfo(hostname, base_port)[0][4]
+            self._ip_addr = addr_tuple[0]
+            self._base_port = addr_tuple[1]
+        except gaierror:
+            raise gaierror("Hostname '%s' could not be resolved" % hostname)
+
+    def is_ipv6(self):
+        return self._ip_addr.rfind(':')
```

Comment:
I think the right way is to avoid detecting IPv6 address that way. ``getaddrinfo`` returns a tuple, from which you could identify if hostname is an IPv4 or 6. Also, it handles all parsing errors, if there is wrong format (or wrong numbers) of ip address.